source/fundamentals/builders/updates.txt

```diff
@@ -11,3 +11,505 @@ Updates Builders
+   1010
+   ----
+   1111
+
+The example above updates the document to the following state:
+
+.. code-block:: json
+   :copyable: false
+      
+   {
+      "_id": 1, 
+      "color": "red", 
+      "qty": 15,
+      "vendor": [ "A", "D", "M" ],
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021
+   }
+
+.. _array_updates:
+
+Array Updates
+-------------
+
+Add to Set
+~~~~~~~~~~
+Use the :java-core-api:`addToSet() <com/mongodb/client/model/Updates.html#addToSet(java.lang.String,TItem)>`
+method to append a value to an array if the value is not already present
+in an update operation. 
+
+The following example adds the value "C" to the array value of
+the ``vendor`` field: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Updates.java
+   :language: java
+   :dedent:
+   :start-after: begin addToSetUpdate
+   :end-before: end addToSetUpdate
+
+The example above updates the document to the following state:
+   
+.. code-block:: json
+   :copyable: false
+      
+   {
+      "_id": 1, 
+      "color": "red", 
+      "qty": 5, 
+      "vendor": [ "A", "D", "M", "C" ],
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021
+   }
+
+Pop
+~~~
+Use the :java-core-api:`popFirst() <com/mongodb/client/model/Updates.html#popFirst(java.lang.String)>`
+method to remove the first element of an array and the
+:java-core-api:`popLast() <com/mongodb/client/model/Updates.html#popLast(java.lang.String)>`
+method to remove the last element of an array in an update operation. 
+
+The following example pops the first element off of the array value
+of the ``vendor`` field:  
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Updates.java
+   :language: java
+   :dedent:
+   :start-after: begin popFirstUpdate
+   :end-before: end popFirstUpdate
+
+The example above updates the document to the following state:
+   
+.. code-block:: json
+   :copyable: false
+      
+   {
+      "_id": 1, 
+      "color": "red", 
+      "qty": 5, 
+      "vendor": [ "D", "M" ],
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021
+   }
+
+Pull All
+~~~~~~~~
+Use the :java-core-api:`pullAll() <com/mongodb/client/model/Updates.html#pullAll(java.lang.String,java.util.List)>`
+method to remove all instances of a value from an existing array in
+an update operation. 
+
+The following example removes vendor "A" and "M" from the ``vendor`` array: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Updates.java
+   :language: java
+   :dedent:
+   :start-after: begin pullAllUpdate
+   :end-before: end pullAllUpdate
+
+The example above updates the document to the following state:
+   
+.. code-block:: json
+  :copyable: false
+      
+   {
+      "_id": 1, 
+      "color": "red", 
+      "qty": 5, 
+      "vendor": [ "D" ],
+      "lastModified" = Fri Mar 05 00:00:00 EST 2021
+   }
+
+Pull
+~~~~
+Use the :java-core-api:`pull() <com/mongodb/client/model/Updates.html#pull(java.lang.String,TItem)>`
+method to remove all instances of a value from an existing array in
```

Comment:
Does it remove all or just the first?  As is, the description of this method is the same as the description for pullAll. (I think the Javadoc has the same confusion, but it would be good to clarify here and perhaps also fix the Javadoc)

Reply:
So the difference is that pull only removes one value, but pullAll can do multiple. I didn't properly convey it in this PR, so I'll be updating it

Reply:
What did you end up doing with this one?

Reply:
I fixed the wording for pull to be "of a value" and pullAll to be "of values" since that's how they differ.